alkira/resource_alkira_service_f5_lb_helper.go

```diff
@@ -163,59 +163,46 @@ func deflateF5SegmentOptions(in alkira.F5SegmentOption, m interface{}) ([]map[st
 	return segmentOptions, nil
 }
 
-func setF5Instances(d *schema.ResourceData, c []alkira.F5Instance) []map[string]interface{} {
+// Set `instance` block from the API response, except the creds.
+func setF5Instances(d *schema.ResourceData, ins []alkira.F5Instance) []map[string]interface{} {
 	var instances []map[string]interface{}
 
-	for _, instance := range d.Get("instances").([]interface{}) {
-		tfInstance := instance.(map[string]interface{})
-
-		for _, apiInstance := range c {
-			if tfInstance["id"].(int) == apiInstance.Id || tfInstance["name"].(string) == apiInstance.Name {
-				instanceStruct := map[string]interface{}{
-					"name":                       apiInstance.Name,
-					"id":                         apiInstance.Id,
-					"credential_id":              apiInstance.CredentialId,
-					"registration_credential_id": apiInstance.RegistrationCredentialId,
-					"license_type":               apiInstance.LicenseType,
-					"version":                    apiInstance.Version,
-					"hostname_fqdn":              apiInstance.HostNameFqdn,
-					"deployment_option":          apiInstance.Deployment.Option,
-					"deployment_type":            apiInstance.Deployment.Type,
-					"f5_username":                tfInstance["f5_username"].(string),
-					"f5_password":                tfInstance["f5_password"].(string),
-					"f5_registration_key":        tfInstance["f5_registration_key"].(string),
-				}
-				instances = append(instances, instanceStruct)
-				break
+	for _, in := range ins {
+		// fetch the creds from the terraform state.
+		var f5Username string
+		var f5Password string
+		var f5RegistrationKey string
+		for _, value := range d.Get("instance").([]interface{}) {
+			cfg := value.(map[string]interface{})
+			if cfg["id"].(int) == in.Id || cfg["name"].(string) == in.Name {
+				f5Username = cfg["f5_username"].(string)
+				f5Password = cfg["f5_password"].(string)
+				f5RegistrationKey = cfg["f5_registration_key"].(string)
 			}
 		}
-	}
-
-	for _, apiInstance := range c {
-		new := true
-
-		for _, instance := range d.Get("instances").([]interface{}) {
-			configInstance := instance.(map[string]interface{})
+		instance := map[string]interface{}{
+			"name":                       in.Name,
+			"id":                         in.Id,
+			"credential_id":              in.CredentialId,
+			"registration_credential_id": in.RegistrationCredentialId,
+			"license_type":               in.LicenseType,
+			"version":                    in.Version,
+			"hostname_fqdn":              in.HostNameFqdn,
+			"deployment_option":          in.Deployment.Option,
+			"deployment_type":            in.Deployment.Type,
+		}
 
-			if configInstance["id"].(int) == apiInstance.Id || configInstance["name"].(string) == apiInstance.Name {
-				new = false
-				break
-			}
+		if f5RegistrationKey != "" {
```

Comment:
At this point, we don't need to check the value of those hidden fields anymore, right? If TF state has not this instance, we could just set it to empty...  so there is no need for check at all. It won't matter anyways.